src/content/reference/react-markup/renderToHTML.md

```diff
@@ -0,0 +1,74 @@
+---
+title: renderToHTML
+canary: true
+---
+
+
+<Intro>
+
+`renderToHTML` renders a React tree to non-interactive HTML.
+
+```js
+const stream = renderToHTML(reactNode, options?)
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `renderToHTML(reactNode, options?)` {/*rendertohtml*/}
+
+You can call `renderToHTML` to render a non-interactive tree of React components to HTML.
+By default, it supports shared components and built-in components.
+Server Components are only allowed if used in a React Server environment (e.g. in Server Actions).
+You can also use it during Client-Side Rendering but only without Server Components.
+
+When a `<html>` tag is rendered, `renderToHTML` will automatically add `<!DOCTYPE html>` doctype.
+
+```js
+import { experimental_renderToHTML as renderToHTML } from 'react-markup';
+
+const markup = await renderToHTML(<App />);
+```
+
+#### Parameters {/*parameters*/}
+
+* `reactNode`: A React node you want to render to HTML. For example, a JSX element like `<App />`.
+
+* **optional** `options`: An object for server render.
+  * **optional** `identifierPrefix`: A string prefix React uses for IDs generated by [`useId`.](/reference/react/useId) Useful to avoid conflicts when the markup is embedded in or combined with other markup that is rendered by React.  
+  * **optional** `onError`: A callback that fires whenever there is a server error. By default, this only calls `console.error`. If you override it to [log crash reports,](#logging-crashes-on-the-server) make sure that you still call `console.error`. This is different to the rejection reason of the created Promise since it'll include the parent component stack.
+  * **optional** `signal`: An [abort signal](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal) that lets you [abort rendering](#aborting-server-rendering). The original Promise will reject.
+
+#### Returns {/*returns*/}
+
+`renderToHTML` returns a Promise that will resolve with the HTML string of the rendered React tree.
+
+#### Caveats {/*caveats*/}
+
+* Will throw when Client APIs (e.g. `useState` or `useEffect`) are used.
+
+---
+
+## Usage {/*usage*/}
+
+### Rendering a React tree as an Email {/*rendering-a-react-tree-as-an-email*/}
+
+Await the call of `renderToHTML` :
+
+```js {7}
+import { experimental_renderToHTML as renderToHTML } from 'react-markup';
```

Comment:
BTW there seems to be a package with the exact name already? https://www.npmjs.com/package/react-markup

Reply:
also saw that it it looks like @eps1lon took it over

![CleanShot 2024-08-14 at 17 42 08](https://github.com/user-attachments/assets/8af4f3c4-ab37-4904-985e-5bc458ef72d5)

Reply:
Published a new major version as a placeholder to avoid confusion: https://www.npmjs.com/package/react-markup

Reply:
Also I see that the versioning starts with 2.0.0. It was whole different package before that? Is it possible to remove/unpublish the older packages before the downloads kick in? Because in ESLint npm didn't allow us to un publish package when it start getting downloads. They informed us to mark it as deprecated instead.

Reply:
We deprecated earlier versions. I don't think we should unpublish. Even though it has only 3 weekly downloads, people were still using it so it feels overly aggressive to remove references considering the previous code is harmless.